### PR TITLE
chore: Improves the layout of the VizTypeGallery component

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Improves the layout of the `VizTypeGallery` component.
- Vertical aligns selectors to the center
- Adds top margin to the search bar
- Sets the charts panel scroll to auto
- Increases the panel padding
- Replaces the `Select` component
- Decreases the width of the select and adds the hint to the same line
- Sets a `border-radius` to match the rest of the application

@junlincc @rusackas @suddjian 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-07-27 at 10 45 44 AM

Screen Shot 2021-07-27 at 10 43 02 AM

### TESTING INSTRUCTIONS
- Click on Add chart
- Check the changes on the screen

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/addSlice/AddSliceContainer.tsx [hunk 1]
```diff
@@ -115,31 +142,21 @@ export default class AddSliceContainer extends React.PureComponent<
...
+            header={t('Choose a dataset')}
+            onChange={this.changeDatasource}
+            options={this.props.datasources}
+            placeholder={t('Choose a dataset')}
+            value={this.state.datasourceValue}
+          />
+          <span>
             {t(
-              'If the dataset you are looking for is not available in the list, follow the instructions on how to add it in the Superset tutorial.',
+              "If you can't find the dataset, follow the instructions on how to add it in the Superset tutorial.",
```
Comment (rusackas) [inline superset-frontend/src/addSlice/AddSliceContainer.tsx:159]:
```suggestion
              "If you can't find a dataset, follow the instructions on how to add it in the Superset tutorial.",
```

File: superset-frontend/src/addSlice/AddSliceContainer.tsx [hunk 2]
```diff
@@ -115,31 +142,21 @@ export default class AddSliceContainer extends React.PureComponent<
...
-          </div>
-          <span className="text-muted">
+        <div className="dataset">
+          <Select
+            ariaLabel={t('Dataset')}
+            name="select-datasource"
+            header={t('Choose a dataset')}
+            onChange={this.changeDatasource}
+            options={this.props.datasources}
+            placeholder={t('Choose a dataset')}
```
Comment (michael-s-molina) [inline superset-frontend/src/addSlice/AddSliceContainer.tsx:154]:
```suggestion
            placeholder={t('Choose a dataset')}
            showSearch
```

File: superset-frontend/src/addSlice/AddSliceContainer.tsx [hunk 3]
```diff
@@ -115,31 +142,22 @@ export default class AddSliceContainer extends React.PureComponent<
...
+            showSearch
+            value={this.state.datasourceValue}
+          />
+          <span>
             {t(
-              'If the dataset you are looking for is not available in the list, follow the instructions on how to add it in the Superset tutorial.',
+              "If you can't find a dataset, follow the instructions on how to add it in the Superset tutorial.",
```
Comment (suddjian) [inline superset-frontend/src/addSlice/AddSliceContainer.tsx:160]:
```suggestion
              "Instructions to add a dataset are available in the Superset tutorial.",
```

**Review by suddjian — APPROVED** [review]
Great work on polishing the feature! I left a copy suggestion.

**Pull request merged**